cobbler/actions/check.py

```diff
@@ -125,42 +125,41 @@ def check_service(self, status, which, notes=""):
         :param notes: A manual not to attach.
         """
         if notes != "":
-            notes = " (NOTE: %s)" % notes
+            notes = f" (NOTE: {notes})"
         return_code = 0
         if process_management.is_supervisord():
             with ServerProxy("http://localhost:9001/RPC2") as server:
                 process_info = server.supervisor.getProcessInfo(which)
                 if process_info["statename"] != "RUNNING":
-                    status.append("service %s is not running%s" % (which, notes))
+                    status.append(f"service {which} is not running{notes}")
                     return
         elif process_management.is_systemd():
             return_code = utils.subprocess_call(
-                "systemctl status %s > /dev/null 2>/dev/null" % which, shell=True
+                ["systemctl", "status", which], shell=False
```

Comment:
Not really clear here where `stderr` and `stdout` will be forwarded to. Do we need to handle these streams somehow?
The same question to the rest of the `subprocess_call` with `> /dev/null 2>/dev/null`

Reply:
No we don't. Python doesn't capture them, thus it discards it for us, no need to duplicate that effort.

Reply:
That's why I'm asking. Maybe I'm wrong, but it will be emited to stderr/stdout of the main process. Like in this test:
```
> python3
Python 3.10.6 (main, Aug 02 2022, 17:22:31) [GCC] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import subprocess
>>> subprocess.call(["ls", "/tmpa"])
ls: cannot access '/tmpa': No such file or directory
2
```
`ls: cannot access '/tmpa': No such file or directory` was emitted here to `stderr`

Reply:
We are using `subprocess.Popen`, is that maybe behaving differently?

Reply:
As I remember `subprocess.Popen` doing it the same way as `subprocess.Popen` if no any extra options for forwarding the streams were specified.

Reply:
The way I understand it, `subprocess.run` does not capture either `stderr` or `stdout`. What you see @vzhestkov is probably your shell capturing it:
```
% python3 2>/dev/null
>>> import subprocess
>>> subprocess.run(["ls", "foobar"])
CompletedProcess(args=['ls', 'foobar'], returncode=2)
```

And you're right, `subprocess.run` is using `subprocess.Popen` under the hood.

Reply:
So then there is something to do or not? Because Alex example explicitly redirects stderr to `/dev/null`.

Reply:
Nothing to be done, in my example `python3` did not receive `stderr`, the shell did. To get the `stderr` into `python3`, I had to:
```
>>> subprocess.run(["ls", "foobar"], stderr=subprocess.PIPE)
CompletedProcess(args=['ls', 'foobar'], returncode=2, stderr=b"ls: cannot access 'foobar': No such file or directory\n")
```

Reply:
Okay then I will resolve this conversation.